pkgs/build-support/bintools-wrapper/setup-hook.sh

```diff
@@ -65,7 +65,7 @@ do
 done
 
 # If unset, assume the default hardening flags.
-: ${NIX_HARDENING_ENABLE="fortify stackprotector pic strictoverflow format relro bindnow"}
+: ${NIX_HARDENING_ENABLE="@default_hardening_flags_str@"}
```

Comment:
Ugh, this means that changing `bintools-wrapper.passthru.defaultHardeningFlags` triggers a full rebuild of everything -- from `gcc` on up.

Could we move this defaulting logic out of bash code and into Nix code so this doesn't happen?

Reply:
Well here's the fundamental tradeoff of single-sourcing the two defaults. If you're going to have the `NIX_HARDENING_ENABLE` defaults synchronized with the `mkDerivation` defaults, you *have* to generate a new `setup-hook.sh` when those defaults are updated.

The only way I could think of avoiding this would be by passing the "defaults" to the wrapper at runtime through an env var - but that doesn't actually solve the problem, because we'd need defaults to fall back to when *that* env var weren't set (remember, wrapped CCs aren't just used for building nixpkgs packages, it's what a user gets when they ask for `nix-shell -p clang`, so perfectly likely to be called without the "expected" env vars)

Reply:
> If you're going to have the wrapper defaults synchronized with the `mkDerivation` defaults, you _have_ to generate a new `setup-hook.sh` when those defaults are updated.

I guess doing the defaulting in Nix code would only avoid rebuilds for derivations which *don't* use the defaults -- which is a small minority of the cases.

So yeah I guess it doesn't matter.